pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -66,3 +78,66 @@ func GetProjectVolumeLabels(componentName string) map[string]string {
 	}
 	return volumeLabels
 }
+
+// GetSupervisordVolumeLabels returns the label selectors used to retrieve the supervisord volume
+func GetSupervisordVolumeLabels() map[string]string {
+	supervisordLabels := map[string]string{
+		"name": adaptersCommon.SupervisordVolumeName,
+		"type": "supervisord",
+	}
+	return supervisordLabels
+}
+
+// CreateAndInitSupervisordVolume creates the supervisord volume and initializes it with odo init
+func CreateAndInitSupervisordVolume(client lclient.Client) (string, error) {
+	supervisordLabels := GetSupervisordVolumeLabels()
+	supervisordVolume, err := client.CreateVolume(supervisordLabels)
+	if err != nil {
+		return "", errors.Wrapf(err, "Unable to create supervisord volume for component")
+	}
+	supervisordVolumeName := supervisordVolume.Name
+
+	err = StartBootstrapSupervisordInitContainer(client, supervisordVolumeName)
+	if err != nil {
+		return "", errors.Wrapf(err, "Unable to start supervisord container for component")
+	}
+
+	return supervisordVolumeName, nil
+}
+
+// StartBootstrapSupervisordInitContainer ...
+func StartBootstrapSupervisordInitContainer(client lclient.Client, supervisordVolumeName string) error {
+	supervisordLabels := GetSupervisordVolumeLabels()
+
+	image := adaptersCommon.GetBootstrapperImage()
+	command := []string{"/usr/bin/cp"}
+	args := []string{
+		"-r",
+		adaptersCommon.OdoInitImageContents,
+		adaptersCommon.SupervisordMountPath,
+	}
+
+	s := log.Spinner("Pulling image " + image)
```

Comment:
Maybe put this log behind a `glog.v(3)`? Don't think the user needs to see it by default